### PR TITLE
EVG-14357 lookup tagged commit with github API

### PR DESCRIPTION
**Pull request by ablack12**
The webhook information we get from Github is inconsistent; sometimes HeadCommit doesn't exist, sometimes it contains the wrong hash, etc. There's an API call that will get us tag information just using the tag name, but the library doesn't have a method for it so I'm calling it directly.
Tested this by pushing a lightweight tag and an annotated tag on staging, and verifying that a version was created from the right commit.

### CODE REVIEW

File: rest/route/github.go [hunk 1]
```diff
@@ -337,14 +337,29 @@ func (gh *githubHookApi) handleGitTag(ctx context.Context, event *github.PushEve
...
+		return errors.New("error getting github token")
+	}
 	pusher := event.GetPusher().GetName()
 	tag := model.GitTag{
 		Tag:    strings.TrimPrefix(event.GetRef(), refTags),
 		Pusher: pusher,
 	}
 	ownerAndRepo := strings.Split(event.Repo.GetFullName(), "/")
-	hash := event.HeadCommit.GetID()
+	hash, err := thirdparty.GetTaggedCommitFromGithub(ctx, token, ownerAndRepo[0], ownerAndRepo[1], tag.Tag)
```
Comment (malikchaya2) [inline rest/route/github.go:350]: Is there something we can do to make sure that ownerAndRepo[1] won't panic? Maybe check that the array length is at least 2?

Comment (ablack12) [reply rest/route/github.go:350]: This actually does happen in validatePushTagEvent

**Review by malikchaya2 — APPROVED** [review]

File: rest/route/github.go (hunk 1 above)
Comment (john-m-liu) [inline rest/route/github.go:350]: Would it make sense to not do this check if event.HeadCommit.GetID() exists?

**Review by john-m-liu — APPROVED** [review]

File: rest/route/github.go (hunk 1 above)
Comment (ablack12) [reply rest/route/github.go:350]: There have been a couple of instances of the ID at HeadCommit not being what we'd expect. Github doesn't even document HeadCommit so there's really not a guarantee on what we should expect.

**Pull request merged**